datadriven/src/test_data_reader.rs

```diff
@@ -0,0 +1,153 @@
+use crate::line_sparser::parse_line;
+use crate::test_data::TestData;
+use anyhow::Result;
+use std::path::{Path, PathBuf};
+// use for writeln! macro
+use std::fmt::Write;
+use std::iter::Enumerate;
+use std::str::Lines;
+
+pub struct TestDataReader<'a> {
+    source_name: PathBuf,
+    pub data: TestData,
+    scanner: Enumerate<Lines<'a>>,
+    logger: slog::Logger,
+}
+
+impl<'a> TestDataReader<'a> {
+    pub fn new<P>(source_name: P, content: &'a str, logger: &slog::Logger) -> Self
+    where
+        P: AsRef<Path>,
+    {
+        Self {
+            source_name: source_name.as_ref().to_path_buf(),
+            scanner: content.lines().enumerate(),
+            data: TestData::default(),
+            logger: logger.clone(),
+        }
+    }
+
+    pub fn next(&mut self) -> Result<bool> {
+        loop {
+            let line = self.scanner.next();
+            if line.is_none() {
+                return Ok(false);
+            }
+
+            self.data = TestData::default();
+            self.data.pos = format!(
+                "{} : L{}",
+                self.source_name.as_path().display(),
+                line.unwrap().0
+            );
+
+            let mut line = line.unwrap().1.to_string();
+
+            // before argument only (1) comment (2) empty line are accepted
+            if line.starts_with('#') || line.is_empty() {
+                // Skip comment lines.
+                continue;
+            }
+
+            // Support wrapping directive lines using \, for example:
+            // build-scalar \
+            // vars(int)
+            while line.ends_with('\\') {
+                line = line.trim_end_matches('\\').to_string();
+
+                let l = self
+                    .scanner
+                    .next()
+                    .expect("expected argument that is not end with '\\'")
```

Comment:
It doesn't panic in the original implement.

Reply:
This will happen when the testdata is not complete, or there is no argument end without `\`.